### PR TITLE
chore(lint-license): add lint checker

### LINKED ISSUE #4792: [lint-check]: feat add check for Copyright/license header
## What will this achieve?

We are currrently manually checking if the Copyright header is on every file when as part of the release review process, but instead we can check this with a linter automatically like we do in Dotcom: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/tasks/check-license.js

## How will success be measured?

When introducing new code to the repository the check-license linter will run and will fail and not allow user to create a commit if there is no Copyright header at the top of every file

### PR DESCRIPTION
**Pull request by ariellalgilmore**
Closes #4792 

#### What did you change?

added a lint checker to see if a license was in the file and update the year when necessary 

#### How did you test and verify your work?
ran the lint checks locally

### CODE REVIEW

**Review by davidmenendez — APPROVED** [review]
wow this is awesome!

File: scripts/license-text.cjs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright IBM Corp. 2020
```
Comment (elycheea) [inline scripts/license-text.cjs:2]: I’m guessing this should also be 2024?

File: packages/ibm-products/src/components/AboutModal/AboutModal.stories.jsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright IBM Corp. 2020, 2024
+ * Copyright IBM Corp. 2024
```
Comment (elycheea) [inline packages/ibm-products/src/components/AboutModal/AboutModal.stories.jsx:2]: I expected this one to retain the first year as it does in ActionBar 🤔

File: scripts/license-text.cjs (hunk 1 above)
Comment (ariellalgilmore) [reply scripts/license-text.cjs]: updated!

File: packages/ibm-products/src/components/AboutModal/AboutModal.stories.jsx (hunk 2 above)
Comment (ariellalgilmore) [reply packages/ibm-products/src/components/AboutModal/AboutModal.stories.jsx]: updated! was testing in aboutmodal

File: packages/ibm-products/src/components/AboutModal/AboutModal.stories.jsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright IBM Corp. 2024
+ * Copyright IBM Corp. 2020
```
Comment (elycheea) [inline packages/ibm-products/src/components/AboutModal/AboutModal.stories.jsx:2]:
```suggestion
 * Copyright IBM Corp. 2020, 2024
```

**Review by matthewgallo — CHANGES_REQUESTED** [review]

**Review by matthewgallo — DISMISSED** [review]
👏

**Review by davidmenendez — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]
woot

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**